src/buffer.h

```diff
@@ -47,24 +48,24 @@ struct mpd_buffer {
 	/** the actual buffer */
 	unsigned char *data;
 
+	/** buffer allocated */
+	bool data_allocated;
```

Comment:
What's the point of this? And what's the point of setting `data_size` to a non-zero value when nothing has been allocated? I'm puzzled about how complicated you chose to make this commit.

Reply:
So that we can do lazy allocation.
That variable can be removed if we compare `buffer->data == NULL`, but i thought you would not like that

the objective of setting data_size to non-zero was to avoid having two returns in `mpd_buffer_room()`, and having an additional if in `mpd_buffer_make_room()` for having or not allocated yet. The buffer will be allocated once the function calls `mpd_buffer_write` (or `mpd_buffer_make_room()` if the requested size is over `DEFAULT_BUFFER_SIZE`)

Reply:
You can do lazy allocation by setting `data_size` (or `capacity`) to zero, and allocate whenever you need more than the current capacity. No special cases.

The argument in the second paragraph, I don't get it.

Here:
>  This resulted in an increase of code complexity as the code checks for OOM events in more places.

This design choice of yours is causing this complexity. Making special cases, adding more unnecessary states to the buffer object, all of which need to be checked by every caller.

Reply:
I have refactor the code with your suggestions. Could you take another look please?
the commit is https://github.com/MusicPlayerDaemon/libmpdclient/pull/53/commits/ef706f9e741c8778429b0a99776b8a35625da943